.github/workflows/test.yml

```
@@ -0,0 +1,19 @@
+name: Test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.24.0'
```

This format is easier to update.
https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file
```suggestion
          go-version-file: 'go.mod'
```